### PR TITLE
Refactor Guest Window to Class

### LINKED ISSUE #13775: Refactor window to class: Guest
See #13737 for more details.

### LINKED ISSUE #17399: Guest thought "I have the strangest feeling someone is watching me" not appearing in OpenRCT2
### Operating System

Windows 10, 64 bit

### OpenRCT2 build

OpenRCT2, v0.4.0 release

### Describe the issue

When I was playing a bit of vanilla RCT2 and followed around some guests, they think a thought about being followed: "I have the strangest feeling someone is watching me". In the old days that was such a nice thing to see as a child.

However, in OpenRCT2 I could not remember seeing this thought at all; while in vanilla it shows up pretty quickly.
So I think this is a bug in OpenRCT2.

### Area(s) with issue?

_No response_

### Steps to reproduce

1. Open a park
2. Click on a guest to follow him around
3. Within a short time the thought "I have the strangest feeling someone is watching me" should appear, but it doesn't.

### Attachments

_No response_

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Close #13775
Close #17399

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
Will invalidate #17468 and #17218

**Review by Broxzier — COMMENTED** [review]
Tested, and everything seems to work as expected. Left two small notes on hardcoded array sizes.

File: src/openrct2-ui/windows/Guest.cpp [hunk 1]
```diff
@@ -102,1963 +102,1837 @@ static rct_widget window_guest_overview_widgets[] = {
...
 // clang-format off
-static rct_widget* window_guest_page_widgets[] = {
-    window_guest_overview_widgets,
-    window_guest_stats_widgets,
-    window_guest_rides_widgets,
-    window_guest_finance_widgets,
-    window_guest_thoughts_widgets,
-    window_guest_inventory_widgets,
-    window_guest_debug_widgets,
+static constexpr std::array<rct_widget*, 7> _guestWindowPageWidgets = {
```
[the remark this answers is not in the document]
Comment (duncanspumpkin) [reply src/openrct2-ui/windows/Guest.cpp:137]: I've a feeling that the mac compiler will not like that.

[the remark this answers is not in the document]
Comment (duncanspumpkin) [reply src/openrct2-ui/windows/Guest.cpp]: Nope all happy.

**Review by Broxzier — APPROVED** [review]

**Pull request merged**